woocommerce/sourcebuster-js
# Add the single cookie option

Due to some issues with Sourcebuster cookies and some Web Application Firewalls (specifically, using Comodo's WAF ruleset), a handful of the cookies set by Sourcebuster are incorrectly flagged as suspicious, and users are blocked from visiting the host site.

The `sbjs_current` cookie is the only one that's whitelisted, so this PR modifies Sourcebuster behavior with a user flag to store all data as JSON in that single cookie.

It overrides the setter and getter for cookies, and adds them to an object, which is then saved as the single cookie's value. It also erases any other existing cookies (and should retrieve their values if upgrading).

### Testing
1. From `trunk`, build the project and host it, and visit the example site in `/public/index.html`
2. Confirm that the existing Sourcebuster works as expected: correct sources, first and current, session and page visits, new tabs, close and reopen, etc.
    - You can visit the page in incognito windows or erase the cookies to reset the current/first values
    - You can also reduce the cookie lifetime to `.000` and session time to `1` to simulate new sessions more quickly.
4. Build this PR's version, and confirm that
    - There's only one cookie created
    - It's values are updated correctly: correct sources, first and current, session and page visits, new tabs, close and reopen, etc.
 5. Reset the cookies, and disable `single_cookie` (in `index.slim` and rebuild, or in `index.html`).
 6. Confirm everything still works as expected.
 7. Enable `single_cookie`.
 8. Confirm the original cookies are erased
